### PR TITLE
Feat/scroll

### LINKED ISSUE #424: Is it possible to enable horizontal scrolling on the x-axis?
### Question

Hi everyone,

I have a question: is it possible to enable horizontal scrolling on the x-axis? In my case, I have a large array of data, and I want to display 3 items in the chart viewport at a time. If I want to see more, I would like to be able to scroll horizontally. You can refer to the GIF below for reference.

Scrollable

I stuck with that issue for couple of weeks, so anyone has any ides, plz let me know. 
Thank you so much in advance.

### PR DESCRIPTION
**Pull request by keithluchtel**
### Description

https://github.com/user-attachments/assets/b9bae0f4-727d-45b8-bd0b-56113167ad0d

Fixes #424 

#### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### Checklist: (Feel free to delete this section upon completion)

- [x] I have included a [changeset](../CONTRIBUTING.md#changesets) if this change will require a version change to one of the packages.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have run `yarn run check:code` and all checks pass
- [x] I have created a changeset for new features, patches, or major changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by zibs — APPROVED** [review]
awesome, just some small comments

I'd encourage making another example or two if you wanted to as well!

File: lib/src/shared/GestureHandler.tsx [hunk 1]
```diff
@@ -4,16 +4,18 @@ import {
   type GestureType,
 } from "react-native-gesture-handler";
 import {
-  convertToAffineMatrix,
-  convertToColumnMajor,
-  type Matrix4,
+  // convertToAffineMatrix,
```
[the remark this answers is not in the document]
Comment (keithluchtel) [reply lib/src/shared/GestureHandler.tsx:7]: Left for the time being until the referenced GitHub issue below is fixed. Will likely revert back to the original implementation when able.

File: website/docs/cartesian/cartesian-chart.md [hunk 2]
```diff
@@ -88,6 +88,18 @@ An object of shape `{ x?: [number] | [number, number]; y?: [number] | [number, n
 
 For example, passing `domain={{y: [-10, 100]}}` will result in a y-axis with a lower bound of `-10` and an upper bound of `100`. For `domain={{x: [1, 4]}}`, will result in an x-axis contained within those bounds.
 
+### `viewport`
```
[the remark this answers is not in the document]
Comment (keithluchtel) [reply website/docs/cartesian/cartesian-chart.md:91]: I'm going to fix this so it's `[min, max]` format as expected

[the remark this answers is not in the document]
Comment (keithluchtel) [reply website/docs/cartesian/cartesian-chart.md:91]: updated

File: lib/src/types.ts [hunk 3]
```diff
@@ -56,6 +55,10 @@ export type SidedNumber =
 /**
  * Render arg for our line chart.
  */
```
[the remark this answers is not in the document]
Comment (keithluchtel) [reply lib/src/types.ts:57]: fixed

**Comment by cr0nil** [comment]
@keithluchtel How can I reduce the padding at the end of the gridlines, the data ends at 30 and I can move the XAxis all the way to the left side of the view and there is empty space left with no Y lines or data, it should be adjacent to the right side.

**Pull request merged**